### PR TITLE
Dockerize

### PR DESCRIPTION
**Pull request by toliger**
Considering the others issues on docker, I did a concise multi-arch Dockerfile wich work for everyone.

https://hub.docker.com/r/oligertimothee/healthchecks

### CODE REVIEW

**Comment by joshuaavalon** [comment]
If someone is looking for a Docker image, there is an active maintained one from Linux Server.
See [DockerHub](https://hub.docker.com/r/linuxserver/healthchecks) or [GitHub](https://github.com/linuxserver/docker-healthchecks).

**Comment by SuperSandro2000** [comment]
> If someone is looking for a Docker image, there is an active maintained one from Linux Server.
> See [DockerHub](https://hub.docker.com/r/linuxserver/healthchecks) or [GitHub](https://github.com/linuxserver/docker-healthchecks).

I wouldn't trust them with anything. They have really poor support and a few of their images are basically broken by design and they don't care.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,54 @@
+ARG BUILD_DATE=""
+ARG ARCH=amd64
+ARG PYTHON_VERSION=3
+
+
+# First stage
+FROM docker.io/${ARCH}/python:${PYTHON_VERSION}-alpine3.8 as builder
```
Comment (SuperSandro2000) [inline Dockerfile]: This should be shortened to
``FROM python:3.8-rc-alpine``

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,54 @@
+ARG BUILD_DATE=""
```
Comment (SuperSandro2000) [inline Dockerfile]: ARG BUILD_DATE

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ARG ARCH=amd64
+ARG PYTHON_VERSION=3
```
Comment (SuperSandro2000) [inline Dockerfile]: Would be more useful is this would 3.8 or 3.7 or something like that.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+# First stage
+FROM docker.io/${ARCH}/python:${PYTHON_VERSION}-alpine3.8 as builder
+
+# Install deps
+COPY requirements.txt /tmp
+
+RUN apk add --no-cache \
+    build-base \
+    postgresql-dev \
+    linux-headers
```
Comment (SuperSandro2000) [inline Dockerfile]: We only need ``make``, ``g++`` and ``musl-dev`` IIRC.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+USER healthchecks 
+
+EXPOSE 8000/tcp
+
+ARG SYNAPSE_VERSION
+ARG PYTHON_VERSION
+ARG BUILD_DATE
+
+CMD ["uwsgi", "--enable-threads", "uwsgi.ini"]
```
Comment (SuperSandro2000) [inline Dockerfile]: ``enable-threads`` can go into the ``uwsgi.ini``

File: README.md [hunk 6]
```diff
@@ -281,6 +281,33 @@ test them on a copy of your database, not on the live database right away.
...
+
+## Deployment
+
+### Docker
+
+To run the app, you can:
+
+    $ docker build -t healthchecks:latest .
+
+    $ docker volume create \
```
Comment (SuperSandro2000) [inline README.md]: Necessary. You can simply mount ``/data`` to a volume.

File: uwsgi.ini [hunk 7]
```diff
@@ -0,0 +1,8 @@
+[uwsgi]
+http-socket = :8000
+enable-threads
```
Comment (SuperSandro2000) [inline uwsgi.ini]: To work this probably needs ``= true``

File: uwsgi.ini [hunk 8]
```diff
@@ -0,0 +1,8 @@
...
+plugin = python3
+module = hc.wsgi:application
+static-map = /static=static-collected
+static-gzip-dir = static-collected/CACHE
```
Comment (SuperSandro2000) [inline uwsgi.ini]: Does hc put ``.gz`` files in their? My installation lags them.

File: Dockerfile (hunk 4 above)
Comment (toliger) [reply Dockerfile]: The first stage need some headers. For example `linux/limits.h` is not include in musl-dev.
I think `make` and `g++` are useless.

File: README.md (hunk 6 above)
Comment (toliger) [reply README.md]: yes

**Comment by BobCashStory** [comment]
and now ?

**Comment by overclockedllama** [comment]
why no docker support :(

**Comment by mountaindude** [comment]
There is a Docker image here:
https://hub.docker.com/r/linuxserver/healthchecks

Been using it during several months now, works really well.

**Comment by overclockedllama** [comment]
ooh thanks! will try it out. 
Would be nice if the project maintainers provided docker support tho 
@cuu508 @everyone_else
# 😉 👀

**Comment by overclockedllama** [comment]
@mountaindude would you be willing to share a docker-compose file? I have been having some trouble setting up SMTP and I also have no twillio access at all in my integrations

**Comment by mountaindude** [comment]
@soneji  Sure, here is my (slightly anonymized) docker-compose file.

Note that an external network is used, with both healthchecks and database using that network (an easy way to have them find each other). You need to create the external network (called "mynetwork" below) using the "docker network create" command. 

Also. note that if using GMail you need to set up an application password in your GMail config pages at Google. 


```
version: "2.4"
services:
  healthcheck:
    image: linuxserver/healthchecks
    container_name: healthcheck
    depends_on:
      healthcheck-postgres:
        condition: service_started
    environment:
      - PUID=1000
      - PGID=1000
      - SITE_ROOT=https://healthcheck.mydomain.net
      - SITE_NAME=My fancy healthcheck service
      - MASTER_BADGE_LABEL=Healthchecks rocks!
      - REGISTRATION_OPEN=False
      - DB=postgres
      - DB_HOST=healthcheck-db
      - DB_PORT=5432
      - DB_NAME=hc
      - DB_USER=hc
      - DB_PASSWORD=superdupersecret
      - DB_SSLMODE=prefer
      - DEFAULT_FROM_EMAIL=healthcheckm@mydomain.com
      - EMAIL_HOST=smtp.gmail.com
      - EMAIL_PORT=587
      - EMAIL_HOST_USER=joetheuser@mydomain.com
      - EMAIL_HOST_PASSWORD=abcdefgh1234567
      - EMAIL_USE_TLS=True
      - ALLOWED_HOSTS=*
      - SUPERUSER_EMAIL=annatheuser@mydomain.com
      - SUPERUSER_PASSWORD=evenmoretopsecret
      - METRICS_KEY=keep-this-secret-too
    volumes:
      - ./config:/config
    ports:
      - 8000:8000
    restart: always
    networks:
      - healthcheck
    logging:
      driver: "json-file"
      options:
        max-file: "5"
        max-size: "5m"

  healthcheck-db:
    image: postgres:9.6
    container_name: healthcheck-postgres
    restart: always
    ports:
      - "5434:5432"
    environment:
      - POSTGRES_USER=hc
      - POSTGRES_PASSWORD=superdupersecret
      - POSTGRES_DB=hc
    volumes:
      - ./data:/var/lib/postgresql/data
    networks:
      - healthcheck
    logging:
      driver: "json-file"
      options:
        max-file: "5"
        max-size: "5m"

networks:
  healthcheck:
    external:
      name: mynetwork
```

**Comment by soneji** [comment]
Thank you Mr @mountaindude!
It turns out the issue I was having was me using the wrong SMTP port! All working now, very grateful

**Pull request closed without merging**